### DISCUSSION TITLE
ISO knowledge of `zone.detached()`

### DISCUSSION THREAD

**Discussion by tgrecojs** [Agoric/agoric-sdk#9210] [Q&A]
I'm currently utilizing the zone API to establish a durable zone that I can refer to when creating a ﻿`setStore`, `﻿mapStore`, etc. This method of store creation seemingly matches my use-case—at least, that's the assumption I've made. Upon encountering the use of ﻿detached in the router implementation found in `﻿@agoric/network`, I realized my knowledge regarding the use of ﻿zone.detached() was nonexistent, thus leading me to open this discussion.

Below is the code from the `@agoric/network` that led me to create this.

https://github.com/Agoric/agoric-sdk/blob/c74c62863bb9de77e66f8d909058804912e72528/packages/network/src/router.js#L108-L138

I'm curious to understand more about ﻿zone.detached(), how stores cultivated with detached zones differ from those shaped with the top-level ﻿`setStore`, `﻿mapStore`, and the circumstances in which it would be apt to opt for one technique over the other.

Keywords: zone, setStore, mapStore, storage

**Comment by dckc** [comment]
@michaelfig help me out here? what does `detached()` do?

**Reply by michaelfig** [reply]
Using the top-level `mapStore`, etc, creates a singleton store that is assigned to the label (here `'prefix'`) in the backing store (such as the durable baggage), and preserved in future incarnations.

A `detached().mapStore`, on the other hand, does not reserve the label, nor populate it automatically, so it is essentially anonymous.  So, to preserve it, you need to connect it to the backing store some other way, like putting it in another store or in an exo object's state.

`detached()` was created to avoid having to manage a dynamic set of reserved labels, leaving the organization to the programmer instead of the default guard rails that zones (and their underlying `makeOnce`) give you.
